### PR TITLE
Normalising the helm service account tests

### PR DESCRIPTION
**Pull request by amoghrajesh**
A small refactor where I noticed that the service account tests were part of the main test classes for some of the helm test files while not for other. Normalised this behaviour.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@hussein-awala @jedcunningham @potiuk this is a minor refactor PR i came across and wanted to fix. Take a look when you have some time

File: dev/breeze/README.md [hunk 1]
```diff
@@ -52,6 +52,6 @@ PLEASE DO NOT MODIFY THE HASH BELOW! IT IS AUTOMATICALLY UPDATED BY PRE-COMMIT.
 
 ---------------------------------------------------------------------------------------------------------
 
-Package config hash: c2bade421865f94fb91f652317d76f3686ad017067349265d8a1d465bbfc2f0796d570bf4777acfb31a7703f1ccb6b0a63c655b97cec9eb9fdeeb1676c804c02
```
Comment (jedcunningham) [inline dev/breeze/README.md]: Why is this in here?

Comment (amoghrajesh) [reply dev/breeze/README.md]: Ah, not sure how it got added. I will revery this change

Comment (amoghrajesh) [reply dev/breeze/README.md]: @jedcunningham I am not sure why but looks like my breeze is forcing this change in and the pre commits fail. Maybe there is a reason for this?

**Comment by potiuk** [comment]
Likely you have some of the breeze setup.py /pyproject.toml / setup.cfglocally modified when you run pre-commit. For some reason breeze will see them as changed even if in your commit they are not changed

**Review by hussein-awala — APPROVED** [review]
The breeze package config hash was already updated in [this commit](https://github.com/apache/airflow/commit/ed000264178056ec3c0f8f96ffe14eba97e2ddd2), I had to rebase main and force push to remove the change from this PR.

LGTM

**Comment by potiuk** [comment]
Ah yeah. ABR. (A)lways (B)e (R)ebased.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Though, I noticed one thing - it seems that those "test-only" changes did NOT trigger helm tests. I will fix it in a separate PR (our selective tests were too selective in this case) so I applied "full tests needed" label and closed/reopened to rebuild it.

**Comment by potiuk** [comment]
Fix to selective checks here: https://github.com/apache/airflow/pull/32941

**Pull request merged**